### ISSUE TITLE
Amount of lines shown in IE/Firefox is 1 less than clamp prop value

### ISSUE THREAD

**Issue by thomasbottonari** [CezaryDanielNowak/React-dotdotdot#10]
This component is working well for me in Chrome/Safari, but when I test in Firefox (for Mac) or IE11, it is truncating after 1 line instead of 2 when I have a clamp set to 2. In the meantime, I am using browser detection to add 1 to the clamp value when in either of those two browsers, but a fix would be ideal.

**Comment by kmcq** [comment]
@kkwiatkowski #11 did not fix this issue. Do you have any ideas what would cause it? Thanks for adding IE and Firefox support!

**Comment by kieranmann** [comment]
@thomasbottonari I also experienced this issue on a project I am working on. It was caused by the line-height not being equal to the height of the block. I would suggest increasing the line-height on the block so that the following is true. line-height \* lines = height

**Comment by puranjayjain** [comment]
@CezaryDanielNowak can we have a new release after #11 ?

**Comment by Muskos** [comment]
+1 IE/Firefox unstable

**Comment by Muskos** [comment]
I caugth it. Dotdotdot work well in one case. I use it in creacct class and it work well. When I included it in simple function it didn't work.

**Comment by ogirniak** [comment]
+1 IE/Firefox not work(
@Muskos can you please provide example when Dotdotdot works for you in IE/Firefox?

**Comment by Muskos** [comment]
@ogirniak I create component  like this. 
```
class Content extends React.Component {
	render() {
	       return (
			<div>
				<Dotdotdot clamp={2}>
					{content.title}
				</Dotdotdot>
			</div>
		)
	}
}
```

**Comment by ogirniak** [comment]
Unfortunately it's not working for me, but maybe the reason is that I'm trying to do it inside ag-grid.
Thank you!

**Comment by makker** [comment]
I had the same issue. In my case I set the clamp value to 3, but in Edge and FF show only 2 lines. IE 11 and other browsers showed 3 lines. I had my CSS line height set in em. Changed the line height unit to px and all browsers started working fine.

**Comment by 2781** [comment]
I had the same issue FF and IE show only 1 lines

 Dotdotdot clamp=3
h2 {contentTitle} /h2
/Dotdotdot

**Comment by nikita-zadorin** [comment]
For me it works only if I add style with line-height on <Dotdotdot>:

const styles = theme => ({
  itemLine: {
    lineHeight: '22px',
  },
})

<Dotdotdot clamp={1} className={classes.itemLine}>
  <Typography className={classes.itemLine}>
    some text
  </Typography>
</Dotdotdot>

**Comment by nikita-zadorin** [comment]
For me it works in FF only if I add style with line-height on Dotdotdot:
```
const styles = theme => ({
  itemLine: {
    lineHeight: '22px',
  },
})

<Dotdotdot clamp={1} className={classes.itemLine}>
  <Typography className={classes.itemLine}>
    some text
  </Typography>
</Dotdotdot>

**Comment by lhguerra** [comment]
Any updates on this, guys? I have the same issue as @nikita-zadorin, and it is only fixed with line-height set to 1.

Ps.: not only in Firefox.

Edit: I got it. I was passing the title element as a child to the DotDotDot component, like this:
`<DotDotDot clamp={2} children={<Title>{title}</Title>} />`

It works fine when I add just the text to DotDotDot:
`<Title children={<DotDotDot clamp={2}>{title}</DotDotDot>} />`

Just a misuse from one who was used to the jQuery lib :joy:

**Comment by LightOri** [comment]
Seems this issue was fixed in version 1.3.1
https://github.com/CezaryDanielNowak/React-dotdotdot#changelog

> round line-height value from computed float value - IE11 fix

I don't face this issue anymore.

**Comment by CezaryDanielNowak** [comment]
![baby-success-meme-png-3](https://user-images.githubusercontent.com/621293/61472045-f358be00-a983-11e9-9fe6-70ddad74e09b.png)
